diff --git a/dist/esm/index.js b/dist/esm/index.js
index d349bf13ef57b4fa3c9226cb7432642cba1371f6..1fc133d89e20a45a67cb0f03793885be5bfeb6b8 100644
--- a/dist/esm/index.js
+++ b/dist/esm/index.js
@@ -233,7 +233,7 @@ const CONFIG_TYPES = new Set(["array", "function"]);
  * Fields that are considered metadata and not part of the config object.
  * @type {Set<string>}
  */
-const META_FIELDS = new Set(["name", "basePath"]);
+const META_FIELDS = new Set(["name", "basePath", "index"]);
 
 /**
  * A schema containing just files and ignores for early validation.
