diff --git a/dist/cjs/index.cjs b/dist/cjs/index.cjs
index cc033c315239251752f4a2feb3d7546ef0d24013..356fc93fe59e43794bb84ea61fd18a45ced0df8f 100644
--- a/dist/cjs/index.cjs
+++ b/dist/cjs/index.cjs
@@ -241,7 +241,7 @@ const CONFIG_TYPES = new Set(["array", "function"]);
  * Fields that are considered metadata and not part of the config object.
  * @type {Set<string>}
  */
-const META_FIELDS = new Set(["name"]);
+const META_FIELDS = new Set(["name", "index"]);
 
 /**
  * A schema containing just files and ignores for early validation.
diff --git a/dist/esm/index.js b/dist/esm/index.js
index 1e68f33e3d641d856ab25b8206be085b6c205067..7dd46e0b8872078bba033ac75c79a9046020d589 100644
--- a/dist/esm/index.js
+++ b/dist/esm/index.js
@@ -221,7 +221,7 @@ const CONFIG_TYPES = new Set(["array", "function"]);
  * Fields that are considered metadata and not part of the config object.
  * @type {Set<string>}
  */
-const META_FIELDS = new Set(["name"]);
+const META_FIELDS = new Set(["name", "index"]);
 
 /**
  * A schema containing just files and ignores for early validation.
